test/unit/test_interface.py

```diff
@@ -102,3 +105,250 @@ def test_abstract_unpaired_data_loader():
     assert data_loader.moving_image_shape == image_shape
     assert data_loader.fixed_image_shape == image_shape
     assert data_loader.num_samples is None
+
+
+def test_generator_data_loader(caplog):
+    """
+    Test the functions in GeneratorDataLoader
+    """
+    generator = GeneratorDataLoader(labeled=True, num_indices=1, sample_label="all")
+
+    # test properties
+    assert generator.loader_moving_image is None
+    assert generator.loader_moving_image is None
+    assert generator.loader_moving_image is None
+    assert generator.loader_moving_image is None
+
+    # not implemented properties / functions
+    with pytest.raises(NotImplementedError):
+        generator.sample_index_generator()
+
+    # implemented functions
+    # test get_Dataset
+    dummy_array = np.random.random(size=(100, 100, 100)).astype(np.float32)
+
+    # mock generator
+    sequence = [
+        dict(
+            moving_image=dummy_array,
+            fixed_image=dummy_array,
+            moving_label=dummy_array,
+            fixed_label=dummy_array,
+            indices=[1],
+        )
+        for i in range(3)
+    ]
+
+    def mock_generator():
+        for el in sequence:
+            yield el
+
+    # inputs, no error means passed
+    generator.data_generator = mock_generator
```

Comment:
here you are using the same generator, which is labeled,
so for testing get dataset, we will need to test unlabeled as well.

https://codecov.io/gh/DeepRegNet/DeepReg/src/122-interface/deepreg/dataset/loader/interface.py

this is saying the line 260 (unlabeled) is not covered